### PR TITLE
Akb 141 check role admin

### LINKED ISSUE #141: feat f1(4): check role and admin and create if does not exists

### PR DESCRIPTION
**Pull request by Aman-codde**
Closes #141

### CODE REVIEW

File: src/server/routers/user-route.ts [hunk 1]
```diff
@@ -13,23 +13,38 @@ export const userRouter = express.Router();
...
 
-        
-userRouter.post("/create-user", async function (req:any, res:any) {
-    const role = await RoleModel.findOne( {name: "BASIC"})
-    const { name, username, email, password } = req.body;
+//create user (basic or admin or pro)
+export async function  createUser(roleName: string, user: any) {
+  const createUserPromise = new Promise(async (resolve, reject) => {
+    const role = await RoleModel.findOne({ name: roleName });
+    console.log(roleName,".....role....", role);
```
Comment (ChelleAmon) [inline src/server/routers/user-route.ts:20]: Unless you want to keep this, I would suggest removing this line.

File: src/server/server.ts [hunk 2]
```diff
@@ -7,6 +7,10 @@ import http from 'http';
 import dotenv from "dotenv";
 import path from 'path';
 import { apiRouter } from './routers/api.routes.js'
+import { RoleModel } from "./schemas/role.schema.js";
+import { UserModel } from "./schemas/user.schema.js";
+import { createUser } from "./routers/user-route.js";
+import { nextTick } from "process";
```
Comment (ChelleAmon) [inline src/server/server.ts]: I would suggest to remove this line to minimize code "noise".

**Review by ChelleAmon — COMMENTED** [review]
For me, this is kind of a big PR to review, because there is a lot happening on these 2 files. Also, I see a lot of console logs, especially in server js. 

Other than that, I really like the implementation of Promises and Async/Await for this issue.

File: src/server/server.ts (hunk 2 above)
Comment (Aman-codde) [reply src/server/server.ts]: good catch! it's deleted now

**Review by ChelleAmon — APPROVED** [review]
Tested the code and it is all working! Adding ADMIN_EMAIL and ADMIN_PWD made this PR work. 

Awesome job here, Aman!

**Review by brycejjacks — APPROVED** [review]
I love at the additions here... im going to follow vinsons rule about if there is more good than bad ... you push it and come back to fix the little things ... there is a lot goin on in the console which is very advantageous, but that is not common practice.. also this was really big .. i know you said you wanted to make it smaller but there are a few scopes here. great job

**Pull request merged**